syntax = "proto3";

package envoy.extensions.custom_cluster_plugins.cluster_fallback.v3;

option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/custom_cluster_plugins/cluster_fallback/v3;cluster_fallbackv3";

message ClusterFallbackConfig {
  message ClusterConfig {
    string routing_cluster = 1;

    repeated string fallback_clusters = 2;
  }

  message WeightedClusterConfig {
    repeated ClusterConfig config = 1;
  }

  oneof config_specifier {
    ClusterConfig cluster_config = 1;

    WeightedClusterConfig weighted_cluster_config = 2;
  }
}
